crates/matrix-sdk/src/room/mod.rs

```
@@ -2777,11 +2777,29 @@ impl Room {
 
     // Tchap-specific : access_rules
     /// Set or update the access rule for this room.
+    /// We get the value of other acces rule properties (visiblity and encrypted)
+    /// to transmit them to the new RoomAccessRulesEventContent in order to not replace
+    /// existing values with default ones.
+    /// This method is called by client only on an existing room (it is not used on room creation).
+    /// So, the 2 other properties already have a meaningfull value.
     pub async fn set_access_rule(
         &self,
         access_rule: AccessRule,
     ) -> Result<send_state_event::v3::Response> {
-        self.send_state_event(RoomAccessRulesEventContent::new(access_rule)).await
+        let (encrypted, visibility) = match self.full_access_rules().await {
+            Err(_) | Ok(SyncOrStrippedState::Sync(SyncStateEvent::Redacted(_))) => (None, None),
```

@MatMaul 
If we can't get the local values for `encrypted` and `visibility`, we continue wit h`None` and `None`, calling `send_state_event` with `(access_rule, None, None)`.

I think the backend will override customized values with default one (because of the None parameters).

Don't we should return an error in this case rather than calling `send_state_event`?

---

I think this is fine : the event should be in the local store, if it's not it's probably that the event is not in the state, so we should consider the default values.

We could try to fetch the event remotely before defaulting out, but unsure it is worth it.
If the state event actually exists remotely and not locally then it is a bug elsewhere in the SDK, since we properly add this event type to the `required_state`.